src/Umbraco.Web.BackOffice/Filters/OutgoingEditorModelEventAttribute.cs

```diff
@@ -25,33 +29,51 @@ private class OutgoingEditorModelEventFilter : IActionFilter
 
             private readonly IBackOfficeSecurityAccessor _backOfficeSecurityAccessor;
 
+            private readonly IEventAggregator _eventAggregator;
+
             public OutgoingEditorModelEventFilter(
                 IUmbracoContextAccessor umbracoContextAccessor,
-                IBackOfficeSecurityAccessor backOfficeSecurityAccessor)
+                IBackOfficeSecurityAccessor backOfficeSecurityAccessor, IEventAggregator eventAggregator)
             {
                 _umbracoContextAccessor = umbracoContextAccessor
                                           ?? throw new ArgumentNullException(nameof(umbracoContextAccessor));
                 _backOfficeSecurityAccessor = backOfficeSecurityAccessor
                                               ?? throw new ArgumentNullException(nameof(backOfficeSecurityAccessor));
+                _eventAggregator = eventAggregator
+                                   ?? throw new ArgumentNullException(nameof(eventAggregator));
             }
 
             public void OnActionExecuted(ActionExecutedContext context)
             {
                 if (context.Result == null) return;
 
                 var umbracoContext = _umbracoContextAccessor.GetRequiredUmbracoContext();
-                var user = _backOfficeSecurityAccessor.BackOfficeSecurity.CurrentUser;
-                if (user == null) return;
+                var currentUser = _backOfficeSecurityAccessor.BackOfficeSecurity.CurrentUser;
+                if (currentUser == null) return;
 
                 if (context.Result is ObjectResult objectContent)
                 {
                     var model = objectContent.Value;
 
-                    if (model != null)
+                    if (model is ContentItemDisplay content)
+                    {
+                        _eventAggregator.Publish(new SendingContentNotification(content, umbracoContext));
+                    }
+                    else if (model is MediaItemDisplay media)
+                    {
+                        _eventAggregator.Publish(new SendingMediaNotification(media, umbracoContext));
+                    }
+                    else if (model is MemberDisplay member)
+                    {
+                        _eventAggregator.Publish(new SendingMemberNotification(member, umbracoContext));
+                    }
+                    else if (model is UserDisplay user)
+                    {
+                        _eventAggregator.Publish(new SendingUserNotification(user, umbracoContext));
+                    }
+                    else if (model is IEnumerable<Tab<IDashboardSlim>> dashboards)
                     {
-                        var args = new EditorModelEventArgs(model, umbracoContext);
-                        EditorModelEventManager.EmitEvent(context, args);
-                        objectContent.Value = args.Model;
+                        _eventAggregator.Publish(new SendingDashboardsNotification(dashboards, umbracoContext));
```

Comment:
I prefer a switch case with pattern matching in this case..

Reply:
@bergmania right you are 😄 I have updated the PR